skrf/network.py

```diff
@@ -2832,33 +2841,41 @@ def crop(self, f_start: float, f_stop: float, unit: str = None) -> None:
-            scaleFactor = Frequency.multiplier_dict[unit.lower()]/self.frequency.multiplier# make a multiplier to put f_start and f_stop in the right units, e.g. 'GHz' -> 'MHz'
-            f_start *=scaleFactor
-            f_stop *=scaleFactor
+        if unit is not None:  # if `unit` is specified, we must retranslate the frequency units
+            scaleFactor = Frequency.multiplier_dict[
+                              unit.lower()] / self.frequency.multiplier  # make a multiplier to put f_start and f_stop in the right units, e.g. 'GHz' -> 'MHz'
+            f_start *= scaleFactor
+            f_stop *= scaleFactor
 
         if f_start > self.frequency.f_scaled.max():
-            raise ValueError("`f_start` was {}, which was larger than the largest frequency in this Network object, which was {}".format(f_start,self.frequency.f_scaled.max()))
+            raise ValueError(
+                "`f_start` was {}, which was larger than the largest frequency in this Network object, which was {}".format(
+                    f_start, self.frequency.f_scaled.max()))
         if f_stop < self.frequency.f_scaled.min():
-            raise ValueError("`f_stop` was {}, which was smaller than the smallest frequency in this Network object, which was {}".format(f_stop,self.frequency.f_scaled.min()))
+            raise ValueError(
+                "`f_stop` was {}, which was smaller than the smallest frequency in this Network object, which was {}".format(
+                    f_stop, self.frequency.f_scaled.min()))
 
-        start_idx,stop_idx = 0,self.frequency.npoints-1 # start with entire frequency range selected
+        start_idx, stop_idx = 0, self.frequency.npoints - 1  # start with entire frequency range selected
 
         if f_start > self.frequency.f_scaled.min():
             start_idx = find_nearest_index(self.frequency.f_scaled, f_start)
-            if f_start > self.frequency.f_scaled[start_idx]: # we do not want the start index to be at a frequency lower than `f_start`
+            if f_start > self.frequency.f_scaled[
+                start_idx]:  # we do not want the start index to be at a frequency lower than `f_start`
                 start_idx += 1
         if f_stop < self.frequency.f_scaled.max():
             stop_idx = find_nearest_index(self.frequency.f_scaled, f_stop)
-            if f_stop < self.frequency.f_scaled[stop_idx]: # we don't want the stop index to be at a frequency higher than `f_stop`
-                stop_idx -=1
-
-        if stop_idx < start_idx :
-            raise ValueError("Stop index/frequency lower than start: stop_idx: {}, start_idx: {}, self.frequency.f[stop_idx]: {}, self.frequency.f[start_idx]: {}"\
-                                .format(stop_idx,start_idx,self.frequency.f[stop_idx],self.frequency.f[start_idx]  ))
+            if f_stop < self.frequency.f_scaled[
+                stop_idx]:  # we don't want the stop index to be at a frequency higher than `f_stop`
```

Comment:
weird formatting

Reply:
@mbe9a Did you use black for formatting?